packages/react/src/components/ExpandCollapsePanel/ExpandCollapsePanel.test.tsx

```diff
@@ -0,0 +1,245 @@
+  expect(screen.getByText('green')).toBeInTheDocument();
+});
+
+test('should passthrough props', () => {
+  render(
+    <ExpandCollapsePanel title="foo">
+      <div data-testid="test-div" />
+    </ExpandCollapsePanel>
+  );
+
+  const testDiv = screen.getByTestId('test-div');
+  const parent = testDiv.parentElement;
+
+  expect(parent).not.toBeNull();
+  expect(parent).toHaveAttribute('title', 'foo');
+});
+
+test('should have hidden content when collapsed', () => {
+  render(
+    <ExpandCollapsePanel>
+      <div data-testid="test-div">foo</div>
+    </ExpandCollapsePanel>
+  );
+
+  expect(isVisible(screen.getByTestId('test-div'))).toBeFalsy();
+});
+
+test('should have visible content when expanded', () => {
+  render(
+    <ExpandCollapsePanel open>
+      <div data-testid="test-div">foo</div>
+    </ExpandCollapsePanel>
+  );
+
+  expect(isVisible(screen.getByTestId('test-div'))).toBeTruthy();
+});
+
+test('should render PanelTrigger', () => {
+  render(
+    <ExpandCollapsePanel>
+      <PanelTrigger data-testid="panel-trigger" />
+    </ExpandCollapsePanel>
+  );
+
+  const trigger = screen.getByTestId('panel-trigger');
+
+  expect(trigger).toBeInTheDocument();
+  expect(trigger).toHaveAttribute('aria-expanded', 'false');
+});
+
+test('should call onToggle when toggled', () => {
+  const handleToggle = jest.fn();
+  render(
+    <ExpandCollapsePanel onToggle={handleToggle}>
+      <PanelTrigger>Click Me</PanelTrigger>
+      <div />
+    </ExpandCollapsePanel>
+  );
+
+  fireEvent.click(screen.getByRole('button', { name: 'Click Me' }));
+
+  expect(handleToggle).toHaveBeenCalledTimes(1);
+});
+
+test('trigger should open panel collapsed panel', async () => {
+  render(
+    <ExpandCollapsePanel animationTiming={1}>
+      <PanelTrigger>Click Me</PanelTrigger>
+      <div data-testid="test-div" />
+    </ExpandCollapsePanel>
+  );
+
+  fireEvent.click(screen.getByRole('button', { name: 'Click Me' }));
+
+  await waitFor(() => {
+    expect(isVisible(screen.getByTestId('test-div'))).toBeTruthy();
+  });
+});
+
+test('trigger should close expanded panel', async () => {
+  const Component = ({ isOpen }: { isOpen: boolean }) => (
+    <ExpandCollapsePanel animationTiming={1} open={isOpen}>
+      <PanelTrigger>Click Me</PanelTrigger>
+      <div data-testid="test-div" />
+    </ExpandCollapsePanel>
+  );
+
+  const { rerender, getByTestId } = render(<Component isOpen={true} />);
+
+  rerender(<Component isOpen={false} />);
+
+  await waitFor(() => {
+    expect(isVisible(getByTestId('test-div'))).toBeFalsy();
+  });
+});
+
+test('should clean up injected styletags', async () => {
+  const cleanup = jest.spyOn(stylesheets, 'removeStyleTag');
+
+  const { unmount } = render(
+    <ExpandCollapsePanel animationTiming={1}>
+      <PanelTrigger>Click Me</PanelTrigger>
+      <div />
+    </ExpandCollapsePanel>
+  );
+
+  fireEvent.click(screen.getByRole('button', { name: 'Click Me' }));
+
+  unmount();
+
+  await waitFor(() => {
```

Comment:
Thanks for catching, i fixed multiple functions appearance under `waitFor`, but i guess we can't use `find*` instead of `waitFor` in many places, because we use specific function `isVisible` to check `className` of the parent, so probably you have any ideas how we can avoid using `waitFor` for such cases?

Reply:
I thought that we could use `.toBeVisible` instead of needing the `isVisible()` check but the CSS for `is--hidden` isn't loaded for these tests.

https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobevisible